src/plugins/expressions/common/expression_functions/types.ts

```diff
@@ -87,6 +94,32 @@ export interface ExpressionFunctionDefinition<
    */
   fn(input: Input, args: Arguments, context: Context): Output;
 
+  /**
+   * migrate function
```

Comment:
I would suggest to include in these docs the order of how migrations/injections/extractions are applied.
Something like:
"This function will be executed for arguments of an expression function before being executed for the current expression function. This means all arguments will already be migrated when this method is called on an expression function"